apps/web/components/StepperFSM/StepperFSM.tsx

```diff
@@ -0,0 +1,461 @@
+  validateStepperConfig,
+  validateStepConfig,
+} from './StepperFSMUtils'
+import { useRouter } from 'next/router'
+import { richText, SliceType } from '@island.is/island-ui/contentful'
+import { useI18n } from '@island.is/web/i18n'
+import { ValueType } from 'react-select'
+import { ParsedUrlQuery } from 'querystring'
+import { isRunningOnEnvironment } from '@island.is/shared/utils'
+
+const ANSWER_DELIMITER = ','
+export const STEPPER_HELPER_ENABLED_KEY = 'show-stepper-config-helper'
+
+const STEPPER_HELPER_ENABLED =
+  isRunningOnEnvironment('dev') || isRunningOnEnvironment('local')
+
+interface StepperProps {
+  stepper: Stepper
+  startAgainLabel?: string
+  answerLabel?: string
+  backLabel?: string
+  optionsFromNamespace: { slug: string; data: [] }[]
+}
+
+interface StepOptionSelectItem {
+  label: string
+  value: string
+  transition: string
+}
+
+interface QuestionAndAnswer {
+  question: string
+  answer: string
+  slug: string
+}
+
+const getInitialStateAndAnswersByQueryParams = (
+  stepper: Stepper,
+  stepperMachine: StepperMachine,
+  query: ParsedUrlQuery,
+  activeLocale: string,
+  optionsFromNamespace: StepperProps['optionsFromNamespace'],
+) => {
+  let initialState = stepperMachine.initialState
+  const questionsAndAnswers: QuestionAndAnswer[] = []
+
+  const answerString = (query?.answers ?? '') as string
+
+  // If the query parameter is not a string then we just skip checking further
+  if (typeof answerString !== 'string')
+    return { initialState, questionsAndAnswers }
+
+  const answers = answerString.split(ANSWER_DELIMITER)
+
+  for (const answer of answers) {
+    const stateNode = stepperMachine.states[initialState.value as string]
+    const step = getStepBySlug(stepper, stateNode.config.meta.stepSlug)
+    const stepType = resolveStepType(step)
+
+    if (stepType === STEP_TYPES.ANSWER) break
+
+    const options = getStepOptions(step, activeLocale, optionsFromNamespace)
+    const selectedOption = options.find((o) => o.slug === answer)
+    if (!selectedOption) break
+
+    initialState = stepperMachine.transition(
+      initialState,
+      selectedOption.transition,
+    )
+
+    const stepQuestion = getStepQuestion(step)
+    if (stepQuestion) {
+      questionsAndAnswers.push({
+        question: stepQuestion,
+        answer: selectedOption.label,
+        slug: selectedOption.slug,
+      })
+    }
+  }
+
+  return { initialState, questionsAndAnswers }
+}
+
+const StepperFSMWrapper = (
+  StepperFSMComponent: React.ComponentType<StepperProps>,
+) => {
+  const Component = (props: StepperProps) => {
+    const configErrors = validateStepperConfig(props.stepper)
+    const steps =
+      props.stepper?.steps
+        ?.map((s) => s.slug)
+        ?.map((s) => getStepBySlug(props.stepper, s)) ?? []
+
+    const stepConfigErrors = steps.map((step) => ({
+      step,
+      errors: validateStepConfig(step),
+    }))
+
+    const showStepperConfigHelper = STEPPER_HELPER_ENABLED
```

Comment:
Curious about this. Is there a reason the `showStepperConfigHelper` is declared instead of using `STEPPER_HELPER_ENABLED`?

Reply:
I agree with your comment. We will address this in the next PR related to the Stepper.